.github/workflows/scan-image.yml

```diff
@@ -0,0 +1,54 @@
+name: Scan image for vulnerabilities
+on:
+  workflow_call:
+    inputs:
+      ECR_REPOSITORY:
+        required: true
+        type: string
+      ECR_REGION:
+        required: true
+        type: string
+    secrets:
+      ECR_REGISTRY:
+        required: true
+      ECR_ROLE_TO_ASSUME:
+        required: true
+
+jobs:
+  scan-image:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # This is required for requesting the JWT
+      contents: read  # This is required for actions/checkout
+    steps:
+      - name: Assume role in Cloud Platform
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.ECR_ROLE_TO_ASSUME }}
+          aws-region: ${{ inputs.ECR_REGION }}
+
+      - name: Login to container repository
+        uses: aws-actions/amazon-ecr-login@v2
+        id: login-ecr
+        with:
+          mask-password: true
+
+      - name: Scan for vulnerabilities (informative, non-breaking)
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: ${{ secrets.ECR_REGISTRY }}/${{ inputs.ECR_REPOSITORY }}:${{ github.sha }}
+          format: 'table'
+          exit-code: '0'
+          ignore-unfixed: false
+          vuln-type: 'os,library'
+          severity: 'UNKNOWN,LOW,MEDIUM,CRITICAL,HIGH'
```

Comment:
Is this supposed to have CRITICAL and HIGH included?

Reply:
What this step does is check for all severities but does not fail the pipeline. When `ignore-unfixed: false` is set to false it will not ignore vulnerabilities that have no fix for them irregardless of their severity.

However on the step `Scan for breaking vulnerabilities`, this will break pipeline if any vulnerabilities are found, we have `ignore-unfixed: true` otherwise the pipeline will break on issues that we do not have a fix for yet.